ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java

```diff
@@ -250,7 +255,25 @@ void onBlock_consensusValidationShouldNotResolveWhenDataAvailabilityFails() {
 
     importBlockWithError(blockAndState, FailureReason.FAILED_DATA_AVAILABILITY_CHECK_NOT_AVAILABLE);
 
-    assertThatSafeFuture(consensusValidationResult.get()).isNotDone();
+    verify(blockBroadcastValidator, never()).onConsensusValidationSucceeded();
+
+    verify(blobSidecarManager).createAvailabilityChecker(blockAndState.getBlock());
+    verify(blobSidecarsAvailabilityChecker).initiateDataAvailabilityCheck();
+    verify(blobSidecarsAvailabilityChecker).getAvailabilityCheckResult();
+  }
+
+  @Test
+  void onBlock_shouldFailWhenBroadcastValidationFails() {
+    setupWithSpec(TestSpecFactory.createMinimalDeneb());
+    final SignedBlockAndState blockAndState = chainBuilder.generateBlockAtSlot(ONE);
+    storageSystem.chainUpdater().advanceCurrentSlotToAtLeast(blockAndState.getSlot());
+
+    when(blockBroadcastValidator.getResult())
+        .thenReturn(SafeFuture.completedFuture(BroadcastValidationResult.GOSSIP_FAILURE));
+
+    importBlockWithError(blockAndState, FailureReason.FAILED_BROADCAST_VALIDATION);
```

Comment:
Maybe importBlockWithExpectedError? current name always confusing me. Sounds like it sets up some when instead of asserting